### PR TITLE
[SB-17] Add Avatar uploading functionality on User profile page

### PR DESCRIPTION
**Pull request by vitalygv**
https://trello.com/c/wliUGkSI/46-sb-17-add-avatar-uploading-functionality-on-user-profile-page

Avatar added
Снимок экрана 2021-06-30 в 16 04 01

Avatar updated
Снимок экрана 2021-06-30 в 00 57 32

Avatar upload works same way via user profile page (Upload avatar button) and via user edit page (remains same avatar if no new file chosen)

### CODE REVIEW

File: app/controllers/users_controller.rb [hunk 1]
```diff
@@ -6,10 +6,17 @@ def show
...
   end
 
   def update
     @user = user
+    # if @user.avatar.attached?
+    #   @user.avatar.purge
+    #   @user.avatar.attach(params[:avatar])
+    # else
+    #   @user.avatar.attach(params[:avatar])
+    # end
```
Comment (B0RlS) [inline app/controllers/users_controller.rb:19]: Do you really need it?

Comment (vitalygv) [reply app/controllers/users_controller.rb:19]: I followed  tips in the end of the article here https://medium.com/@anaharris/how-to-add-image-upload-functionality-to-your-rails-app-9f7fc3f3d042

Comment (vitalygv) [reply app/controllers/users_controller.rb]: > Do you really need it?

No, removed it. It appeared that no need to add anything to controller at all.

**Comment by B0RlS** [comment]
Why are images of different shapes? Could we make them round so that they all look like the second image in the example?

File: app/models/user.rb [hunk 2]
```diff
@@ -26,4 +28,17 @@ def lock!
   def unlock!
     update(locked: false)
   end
+
+  def acceptable_avatar
+    return unless avatar.attached?
+  
+    unless avatar.byte_size <= 1.megabyte
+      errors.add(:avatar, "is too big")
+    end
```
Comment (B0RlS) [inline app/models/user.rb:37]: move it to private method

Comment (vitalygv) [reply app/models/user.rb]: divided validation into two separate methods

**Pull request merged**